File: Implementierung/Backend/src/main/java/edu/kit/informatik/model/userdata/rooms/Position.java
package edu.kit.informatik.model.userdata.rooms;

import edu.kit.informatik.model.DataObject;
import edu.kit.informatik.model.User;
import jakarta.persistence.Entity;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

/**
 * Entität einer Position
 *
 * @author uekai
 * @author ugqbo
 * @version 1.0
 */
@Getter
@NoArgsConstructor
@Entity
public class Position extends DataObject {

    private double xCoordinate;
    private double yCoordinate;
    private double orientation;

    /**
     * Konstruktor zum Erstellen eines Objektes
     * @param xCoordinate X-Koordinate
     * @param yCoordinate Y-Koordinate
     * @param orientation Drehwinkel
     * @param user {@link User}
     * @param createdAt {@link Timestamp} der Erstellung
     * @param updatedAt {@link Timestamp} der letzten Änderung
     */
    public Position(double xCoordinate, double yCoordinate, double orientation, User user,
                    Timestamp createdAt, Timestamp updatedAt) {
        super(user, createdAt, updatedAt);
        this.xCoordinate = xCoordinate;
        this.yCoordinate = yCoordinate;
        this.orientation = orientation;
    }
}

File: Implementierung/frontend/paschi/src/model/userdata/rooms/Position.ts
import {User} from "@/model/User";
import {DataObject} from "@/model/DataObject";

export class Position extends DataObject {

  private readonly _user: User;
  private _xCoordinate: number;
  private _yCoordinate: number;
  private _orientation: number;

  constructor(id: string | undefined, localId: number, user: User, xCoordinate: number, yCoordinate: number, orientation: number) {
    super(id, localId);
    this._user = user;
    this._xCoordinate = xCoordinate;
    this._yCoordinate = yCoordinate;
    this._orientation = orientation;
  }

  get user(): User {
    return this._user;
  }

  get xCoordinate(): number {
    return this._xCoordinate;
  }

  set xCoordinate(value: number) {
    this._xCoordinate = value;
    this.update();
  }

  get yCoordinate(): number {
    return this._yCoordinate;
  }

  set yCoordinate(value: number) {
    this._yCoordinate = value;
    this.update();
  }

  get orientation(): number {
    return this._orientation;
  }

  set orientation(value: number) {
    this._orientation = value;
    this.update();
  }
}
